src/helper.ts

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as gax from 'google-gax';
+import {Descriptors, ClientOptions} from 'google-gax';
+import * as path from 'path';
+import * as gapicConfig from './iam_policy_service_client_config.json';
+import {ProjectIdCallback} from 'google-auth-library';
+import * as protosTypes from '../protos/protos';
+const version = require('../../package.json').version;
+
+export class IamClient {
+  private _descriptors: Descriptors = {page: {}, stream: {}, longrunning: {}};
+  private _innerApiCalls: {[name: string]: Function} = {};
+  private _terminated = false;
+  // tslint:disable-next-line no-any
+  auth: any;
```

Comment:
why `any`? Maybe `gax.GoogleAuth`?

Reply:
it requires initialization for it and there are many required fields for `gax.GoogleAuth`, thought it might be ok to just have `any` here?
```
Type '{}' is missing the following properties from type 'GoogleAuth': isGCE, jsonContent, cachedCredential, getDefaultProjectId, and 29 more.
```

Reply:
But it works in the regular client :) Why is this one different?

Reply:
I think we can init it in constructor for regular client, and previously (when I tried Object.assign(kms_client, iam_client) ), we do not want constructor because that will be dropped, since kms client has constructor() and won't be overwritten by it. But now I will try to make it a constructor, and see if it works, thanks!

Reply:
You can initialize `auth` in the constructor for a regular client, and pass it as a constructor parameter to IAM client. That way you'll be able to use `gax.GoogleAuth` as a type here.

Reply:
sure, changed from`setupIamClient` to `constructor` and initialize the object there.